pkg/function/function.go

```diff
@@ -164,7 +164,7 @@ func (r *Reconciler) resolveSink(ctx context.Context, f *v1alpha1.Function) (*ap
 func (r *Reconciler) reconcileConfigmap(ctx context.Context, f *v1alpha1.Function) (*corev1.ConfigMap, error) {
 	logger := logging.FromContext(ctx)
 
-	expectedCm := resources.NewConfigmap(f.Name+"-"+rand.String(6), f.Namespace,
+	expectedCm := resources.NewConfigmap(fmt.Sprintf("function-%s-%s", f.Name, rand.String(6)), f.Namespace,
```

Comment:
The ConfigMap isn't the issue, the Ksvc is.
Kubernetes will happily accept a name starting with a digit for ConfigMaps, but Knative will not because that name translates to a DNS name eventually.

Reply:
Missed the line 😒